base/sparse/sparsematrix.jl

```diff
@@ -2896,7 +2896,20 @@ function hcat(X::SparseMatrixCSC...)
     SparseMatrixCSC(m, n, colptr, rowval, nzval)
 end
 
-function hvcat(rows::Tuple{Vararg{Int}}, X::SparseMatrixCSC...)
+
+# Sparse/dense concatenation
+
+function hcat(Xin::Union{Matrix, SparseMatrixCSC}...)
+    X = SparseMatrixCSC[issparse(x) ? x : sparse(x) for x in Xin]
```

Comment:
Generators are new syntax that's like comprehensions but creating an iterator instead of an array. Though it looks like hcat'ing a comprehension without a splat doesn't behave quite like I thought it would.